libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt

```diff
@@ -193,12 +201,15 @@ internal class MapboxTripSession(
     /**
      * Start MapboxTripSession
      */
-    override fun start() {
+    override fun start(withTripService: Boolean) {
+        this.withTripService = withTripService
         if (state == TripSessionState.STARTED) {
             return
         }
         navigator.addNavigatorObserver(navigatorObserver)
```

Comment:
```suggestion
    override fun start(withTripService: Boolean) {
    override fun start(withTripService: Boolean) {
        if (state == TripSessionState.STARTED) {
            if (this.withTripService != withTripService) {
                if (withTripService) {
                    tripService.startService()
                } else {
                    tripService.stopService()
                    notification.hide()
                }
            }
            return
        }
        navigator.addNavigatorObserver(navigatorObserver)
```

or something along these lines. This will allow for starting/stopping the service without restarting the session which might be important from the billing standpoint.

Or should we expose `MapboxNavigation#startForegroundService` and `MapboxNavigation#stopForegroundService` after all?

Reply:
> Or should we expose `MapboxNavigation#startForegroundService` and `MapboxNavigation#stopForegroundService` after all?

I'd prefer not. IMO that'd increase the complexity of how to use the SDK as more APIs are exposed (especially these APIs touching the lifecycle)

Reply:
why do we need that, do we expect that Service on\off might be changed in runtime?

Reply:
I guess I am not clear with the use case. Following are the use cases that could happen
- Start a trip session for the first time `withService = false` and before reaching destination i.e. stop session, stop the service while the session continues
- Start a trip for the first time `withService = false`, reach the destination i.e. stop the session, start a new trip session`withService = true`

/cc @LukasPaczos Do we want to address both the use case, or either of them?

Reply:
The current proposal supports both. The use case I was thinking about is:
1. Start a session `withService = false` to just display location.
2. Request a route.
3. Start active guidance with a new route that requires you to have a foreground service running for a good navigation experience.

Currently, this would require the developer to call `stopSession` and then `startSession( withService = true)` which would incur another billing session. If they could just call `startSession(withService = true)` to only start a service, without needing to stop the earlier session first, they wouldn't be billed twice.

Having said that, I think this would be over-optimizing for other problems. Let's leave the setup as-is for now. The only change that has to be made is moving `this.withTripService = withTripService` _after_ the `if (state == TripSessionState.STARTED)` check. Otherwise, the notification can be shown without the service running (from the route progress update).

Reply:
Okay that's a good context. I think for this use just calling `startSession` to start the service without restarting a new session is misleading. As an external developer when I invoke `startSession` or `stopSession` it's clear that that I am starting or stopping it. To start a service without restarting a session feels like to me like we should either have `startForegroundService` `stopForegroundService` or `updateSessionWithService`. 

This way you can start or stop a session that would start or stop a service. If you want to change the behavior of service invoke the other functions. I don't think we are gonna have lifecycle issues here. 

But anyways this is a conversation for later.

Reply:
Another question:
Now that we can start/stop session without starting a service, I think it's a good addition to expose a method in `TripSession` > `isRunningForegroundService()`

I am gonna add it to this PR. Will remove it if you guys think otherwise.
/cc @Guardiola31337 @LukasPaczos

Reply:
Makes sense to me :+1: we'll need to surface this flag in `MapboxNavigation` as well.